apps/api/src/database/repositories/agent-repository.ts

```diff
@@ -295,6 +295,29 @@ async function findByIdImpl(id: string): Promise<SelectAgent | undefined> {
   }
 }
 
+/**
+ * Find multiple agents by their IDs
+ * @param ids Array of agent IDs to search for
+ * @returns Array of agents matching the provided IDs
+ */
+async function findByIdsImpl(ids: string[]): Promise<SelectAgent[]> {
+  try {
+    if (ids.length === 0) {
+      return [];
+    }
+
+    const results = await db
+      .select()
+      .from(agents)
+      .where(inArray(agents.id, ids));
```

Comment:
note: iirc you were hesitant to use `inArray`?

Reply:
Good callout, as long as the array being passed in isn't huge it shouldn't be a real issue.  
With our current data and usage patterns, this wouldn't be more than the number of agents in a competition, which is about 50.